### ISSUE TITLE
New frontend for EvalAI based on Angular 5

### ISSUE THREAD

**Issue by RishabhJain2018** [Cloud-CV/GSoC-Ideas#23]
**Project Title:** New frontend for EvalAI based on Angular 5

**Description:**

EvalAI’s current frontend is setup using Angular 1 which is not maintained by the community actively. Angular in the later versions support really nice features like better SEO, client-side rendering, etc. We want to migrate the current codebase in Angular 5 with a new design and achieve feature-parity. The first half of the summer will focus on adding the existing features from the older version with a new UI, while the latter half will focus on building an exhaustive analytics platform for challenge host and participants. The tasks will also include adding the UI for hosts and participants for reinforcement learning based challenges.

**Deliverable:**

- [ ] Fix the bugs in Home page / Landing page
	- [ ] Change the challenge creation video on the home page
	- [ ] Add a carousel for adding more organizations
	- [ ] Change image on the homepage of the laptop
- [ ] Add feature to disable the challenge phase submission if it is not active and the challenge is active
- [ ] Add feature to edit every challenge detail
- [ ] Add Discussions tab for each challenge
- [ ] Change notification messages and show the response from the API’s.
- [ ] Add feature to view all the submissions by a challenge host
- [ ] Add feature to display the private leaderboard and private challenge phase
- [ ] Add feature to make the challenge public/private
- [ ] Add feature to delete a challenge
- [ ] Show tags if the challenge is approved or published under my challenges
- [ ] Add the challenge analytics dashboard
- [ ] Add Error 500 page
- [ ] Add challenge creation using UI
- [ ] Make the website loading faster
- [ ] Add new theme templates for email verification, success page, password reset page, password reset success
- [ ] Add feature to evaluate submissions by a challenge host
- [ ] Add feature to export all submissions in different formats and by selecting different fields

**Mentor:** Mayank Lunayach @lunayach, Shekhar Prasad Rajak @Shekharrajak Shekharrajak,
 Shivani Prakash @shivaniprakash95 (Design Mentor) , Rishabh Jain @RishabhJain2018 

**Skills:** Angular 5, HTML, CSS, Typescript

**Skill Level:** Medium

**Get started:** Try to fix some issues in EvalAI (note that there are some issues labeled with **GSOC-2019**)

**Tutorials:**

a) [Angular Tutorial](https://angular.io/tutorial)
b) [Angular Basic Application](https://coursetro.com/courses/19/Learn-Angular-5-from-Scratch---Angular-5-Tutorial)

**Important Links:**

* EvalAI-ngx Github repository: [Cloud-CV/EvalAI](https://github.com/Cloud-CV/EvalAI-ngx)
* EvalAI Docs: [http://evalai.readthedocs.io/en/latest](http://evalai.readthedocs.io/en/latest/)
* EvalAI Website: [evalai.cloudcv.org](https://evalai.cloudcv.org)
* GSoC Proposal Template: [Cloud-CV/GSoC-Ideas/wiki/GSOC-2019-Proposal-Template](https://github.com/Cloud-CV/GSoC-Ideas/wiki/GSOC-2019-Proposal-Template)
* Gitter Channel: [gitter.im/Cloud-CV](https://gitter.im/Cloud-CV/)
* Mailing list: [groups.google.com/forum/#!forum/cloudcv](https://groups.google.com/forum/#!forum/cloudcv)

**Comment by rohitjha941** [comment]
Can i work on this

**Comment by Shekharrajak** [comment]
Hi,

I thinnk required skills should be : 

**Skills:** Angular 2/4/5, HTML, CSS, Typescrip

instead of this : 

**Skills:** AngularJS, HTML, CSS, Typescript

Because there is a lot of difference between AngularJS and Angular 2/4/5.

**Comment by knrt10** [comment]
Sounds like a fun project to work on.

**Comment by sanyamdogra** [comment]
Looks interesting!

**Comment by Abhi58** [comment]
Hi, Can i Work on this? I have experience in Full stack development, basically Angular 4, HTML5 & CSS3 and Typescript.

**Comment by sanyamdogra** [comment]
@shivaniprakash95 will we work on EvalAI-ngx Github repository for EvalAI using Angular? For an angular person there are no issues to solve?

**Comment by Shekharrajak** [comment]
I can see enough open issues to fix : https://github.com/Cloud-CV/EvalAI-ngx/issues

**Comment by RishabhJain2018** [comment]
Hi, @Abhi58 @sanyamdogra @knrt10 Please feel free to solve any of the deliverables as an issue and open a PR for the same.

**Comment by Shekharrajak** [comment]
I have few ideas in mind, I want to discuss it here :

- [Angular 8 beta version](https://github.com/angular/angular/releases) has been released, so it would be better we start thinking about the Angular 7 (stable version) project. So either student can upgrade the Application to angular 7 first and then work on deliverables **or** work on deliverables while keeping in mind, Angular 7 upgrade (means if new dependencies are added then it should be compatible with Angular 7). This will make application faster, since we will be using latest versions in all the dependencies and latest angular dependencies as well.

- The project is using [Angular CLI version 1.6](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/package.json#L34) and current latest [stable version is 7.3.6](https://www.npmjs.com/package/@angular/cli). So it is also required along with the Project Angular version we need to upgrade the CLI version as well (it is needed if we want to upgrade to Angular 7). This will be useful for developers, since it came with new commands and better performance.

- I am expecting better [Test Coverage Report](https://angular.io/guide/testing) and Documentation improvement from the Student. So the student should be aware of Angular Testing and PRs should contain some concrete specs for the changes/new component/service throughout the GSoC coding period.

- Smooth transition between pages and sections for better UI/UX.

What do you people think ? Feel free to add your points :) 

Also, @RishabhJain2018 and Utsav, are we adding enhancement tasks of AngularJS - current EvalAI application into this Angular 5 EvalAI project (may be we can add it here as low priority tasks)?

**Comment by RishabhJain2018** [comment]
Thanks, @Shekharrajak for keeping all these points.

> Angular 8 beta version has been released, so it would be better we start thinking about the Angular 7 (stable version) project. So either student can upgrade the Application to angular 7 first and then work on deliverables or work on deliverables while keeping in mind, Angular 7 upgrade (means if new dependencies are added then it should be compatible with Angular 7). This will make application faster, since we will be using latest versions in all the dependencies and latest angular dependencies as well.

I think we should go with the former approach you proposed.

I totally agree with the rest 3 points you added.

> are we adding enhancement tasks of AngularJS - current EvalAI application into 
this Angular 5 EvalAI project (maybe we can add it here as low priority tasks)?

Sure, we can add those.

**Comment by Shekharrajak** [comment]
Hello Students,

I hope you are aware of application submission deadline which is 9th April 2019. But I will recommend you to collect all your thoughts and write it down steps by steps in you draft proposal as soon as possible , so that you can share it with us. 

- You will get reviews and clear cut idea about the project deliverables and mentors expectation.
- It will be helpful to you to check, you are in the right direction (if not then you have time to explore it further).
- You are welcome to add more ideas apart from this deliverables. You can also check out the https://github.com/Cloud-CV/GSoC-Ideas/issues/21 enhancements ideas, once you are done with above deliverables.

**Comment by padegalmadhuri** [comment]
Hi, I am having the experience in the service worker and hosting plan. Can I work on the issue how to load the website faster?

**Comment by Shekharrajak** [comment]
@padegalmadhuri , Nice! You can collaborate with PR https://github.com/Cloud-CV/EvalAI-ngx/pull/101 author , if you want to add other idea/approach, then go fot it.

**Comment by lunayach** [comment]
Sure @padegalmadhuri, Website's loading speed is very important for the project.  Looking forward to your proposal!

**Comment by mallick370** [comment]
https://github.com/mallick370/EvalAI

I just made a prototype of the landing page before starting the project. Please review it and I am eager to know your views

**Comment by Shekharrajak** [comment]
@mallick370 so it is written in pure HTML,CSS and plane JS ?

**Comment by mallick370** [comment]
@Shekharrajak Yes
